### ISSUE TITLE
Bochs command line interface

### ISSUE THREAD

**Issue by stlintel** [bochs-emu/Bochs#160]
The issue was raised here:
https://sourceforge.net/p/bochs/feature-requests/505/

One of the main reasons for Boch's poor usability is the command line parser.

Bochs command line parser assumes, that device parameters that belong together are given as one argument in quotation marks. If they are not given between quotation marks, Bochs will exit with an error message in most times.

Example:
Correct way:

`$ bochs "clock: sync=realtime, time0=local" "com1: enabled=true, mode=null"`
If you do it with quotation marks, it will be interpreted internally as follows:

```
argv[1]="clock: sync=realtime, time0=local";
argv[2]="com1: enabled=true, mode=null";
```
BUT, if you forget the quotations marks and enter just it without them:
`
$ bochs clock: sync=realtime, time0=local com1: enabled=true, mode=null`
Then this will be interpreted as:

```
argv[1]="clock:";
argv[2]="sync=realtime,";
argv[3]="time0=local";
argv[4]="com1:";
argv[5]="enabled=true,";
argv[6]="mode=null";
```
And this bochs can't handle.
It checks argv[1] and throws an error message, because bochs is missing parameters. The command line parser of bochs is not smart enough to understand, that the next arguments argv[2] and argv[3] belong to the first device option and that these have to be combined.
The user shouldn't have to think about how to pass parameters to bochs.

And if you put all expressions between only 2 quotation marks:

`$ bochs "clock: sync=realtime, time0=local com1: enabled=true, mode=null"
`you get:

`argv[1]="clock: sync=realtime, time0=local com1: enabled=true, mode=null";
`Bochs can't handle that either. It will throw an error, because clock: doesn't know the option com1:.

So to sum it up, with a more intelligent parser, bochs would become instantly much more user friendly.

The user shouldn't have to care if the parameters he gives to bochs on the command line level are in quotations marks or not. Boch's parser should be intelligent enough to consider all options with their parameters regardless of whether they are supplied as 1 argument or dozens of arguments.

To achieve this, it would be best to concatenate all argv[1..n] strings into one single string and only then parse that string.

**Comment by vruppert** [comment]
Do you think it's okay now after the modifications in bx_parse_cmdline() ?

**Comment by stlintel** [comment]
Looks better, thanks !
